source/material_model/visco_plastic.cc

```diff
@@ -284,24 +284,27 @@ namespace aspect
 
           // Calculate changes in viscosity with iterative dampening and fill prescribed output.
           // TODO Should the damping be done before compute_viscosity_derivatives in line 254?
-          if (rheology->use_iterative_viscosity_dampening)
-            {
-              // Set up variable to interpolate the viscosity output onto the compositional field viscosity_field.
-              PrescribedFieldOutputs<dim> *prescribed_field_out = out.template get_additional_output<PrescribedFieldOutputs<dim>>();
-
-              if (in.current_cell.state() == IteratorState::valid && prescribed_field_out != NULL)
-                {
-                  const int field_index = this->introspection().compositional_index_for_name("viscosity_field");
-                  const double old_viscosity = in.composition[i][field_index];
-
-                  // Only dampen the viscosity after the first nonlinear iteration.
-                  if (this->get_nonlinear_iteration() > 0)
-                    out.viscosities[i] = rheology->iterative_dampening->calculate_viscosity(old_viscosity, out.viscosities[i]);
-
-                  // Put the new viscosity onto the compositional field.
-                  prescribed_field_out->prescribed_field_outputs[i][field_index] = out.viscosities[i];
-                }
-            }
+          if (in.requests_property(MaterialProperties::viscosity) || in.requests_property(MaterialProperties::additional_outputs))
+            if (rheology->use_iterative_viscosity_dampening)
+              {
+                // Set up variable to interpolate the viscosity output onto the compositional field viscosity_field.
+                PrescribedFieldOutputs<dim> *prescribed_field_out = out.template get_additional_output<PrescribedFieldOutputs<dim>>();
+
+                if (in.current_cell.state() == IteratorState::valid && prescribed_field_out != NULL)
+                  {
+                    const int field_index = this->introspection().compositional_index_for_name("viscosity_field");
+                    const double old_viscosity = in.composition[i][field_index];
+
+                    // Only dampen the viscosity after the first nonlinear iteration.
+                    if (this->simulator_is_past_initialization() == true &&
+                        this->get_timestep_number() > 0 &&
```

Comment:
I originally also asserted that this should only happen after the first time step, but then switched to doing this on all time steps. What is your reasoning for not doing the dampening on the first time step (I don't recall what my logic was)?

Reply:
I figured that at t0 the initial guesses for velocity/strain rate and pressure are so bad that the viscosity is also far off and changes rapidly with each subsequent nonlinear iteration, and therefore averaging those viscosities would make the convergence better. But I don't know if that's true.

On a sidenote, we're aiming at situations where the residual oscillates, right, which usually only happens after a certain number of nonlinear iterations within a timestep. Maybe we can set `this->get_nonlinear_iteration() > 5` (or some other value larger than 0), but do apply the dampening to each time step?

Reply:
> I figured that at t0 the initial guesses for velocity/strain rate and pressure are so bad that the viscosity is also far off and changes rapidly with each subsequent nonlinear iteration, and therefore averaging those viscosities would make the convergence better. But I don't know if that's true.

That makes sense. If needed, we can always add a parameter that allows switching the dampening on/off on the first time step.

> Maybe we can set this->get_nonlinear_iteration() > 5 (or some other value larger than 0), but do apply the dampening to each time step?

Yep, that sounds like a good strategy.

Reply:
Alright, then after we've rebased, I'll switch to applying averaging after the fifth iteration for each timestep, including the first.